functions/shiritori/index.js

```diff
@@ -14,116 +14,104 @@
 
 'use strict'
 
-const kuroshiro = require('kuroshiro')
+const Kuroshiro = require('kuroshiro')
+const KuromojiAnalyzer = require('kuroshiro-analyzer-kuromoji')
 const wanakana = require('wanakana')
 
 const smallKanas = 'ぁぃぅぇぉゃゅょ'
-let kuroshiroLoaded = false
 
 // 辞書のローディング。
-exports.loaded = new Promise((resolve, reject) => {
-  kuroshiro.init(function (err) {
-    if (err) {
-      reject(err)
-      return
-    }
-    kuroshiroLoaded = true
-    resolve()
-  })
-})
-
-// しりとりのルールのチェック。
-exports.check = (word, chain) => {
-  if ((chain === undefined) || (chain.length === 0)) {
-    return true
-  }
-
-  if (kuroshiroLoaded) {
-    word = kuroshiro.toHiragana(word)
-    chain = chain.map((e) => kuroshiro.toHiragana(e))
-  }
+const kuroshiro = new Kuroshiro()
+const loaded = kuroshiro.init(new KuromojiAnalyzer())
 
-  // 漢字からひらがなにする。
-  const wordHira = wanakana.toHiragana(word)
-  const chainHira = chain.map(wanakana.toHiragana)
+class Win extends Error {}
+exports.Win = Win
+class Bad extends Error {}
```

Comment:
Bad isn't directly the opposite of Win. I don't know if that's intended or not.

Reply:
yep, because they don't really act at the same level currently `Bad` is for the `check` function, surfacing that the current word doesn't conform to the shiritori rule. And `Win` surface that the game ended with a winning condition (usually no word left in dict).

Maybe we should resurface the `Bad` exception as `Loose` exception in `interact` to make this clearer?

Reply:
Yeah if you renamed it to `Lose`, it would have better symmetry and may be more clear.

Reply:
Done. Now I'm returning structured enums.